tests/integration/ceph_bucket_notification_test.go

```diff
@@ -122,38 +110,98 @@ func (s *ObjectSuite) TestBucketNotificationsInOrder() {
-			notifications, err := s3client.Client.GetBucketNotificationConfiguration(&s3.GetBucketNotificationConfigurationRequest{
-				Bucket: &bucketname,
-			})
-			if err != nil {
-				logger.Infof("failed to fetch bucket notifications configuration due to %v", err)
-				return false
-			}
-			logger.Infof("%d bucket notifications found in: %v", len(notifications.TopicConfigurations), notifications)
-			for _, notification := range notifications.TopicConfigurations {
-				if *notification.Id == notificationName {
-					return true
-				}
-				logger.Infof("bucket notifications name mismatch %q != %q", *notification.Id, notificationName)
-			}
-			return false
+	t.Run("add new topic, notification to existing obc", func(t *testing.T) {
+		newNotificationName := "new-notification"
+		newTopicName := "new-topic"
+		err := helper.TopicClient.CreateTopic(newTopicName, storeName, httpEndpointService)
+		assert.Nil(t, err)
+		created := utils.Retry(12, 2*time.Second, "topic is created", func() bool {
+			return helper.TopicClient.CheckTopic(topicName)
 		})
 		assert.True(t, created)
-		logger.Info("CephBucketNotification created successfully on bucket")
+		err = helper.NotificationClient.CreateNotification(newNotificationName, newTopicName)
+		assert.Nil(t, err)
+		created = utils.Retry(12, 2*time.Second, "notification is created", func() bool {
+			return helper.NotificationClient.CheckNotification(notificationName)
+		})
+		assert.True(t, created)
+		obc, err := k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Get(ctx, obcName, metav1.GetOptions{})
+		assert.Nil(t, err)
+		obc.Labels[bucketNotificationLabelPrefix+newNotificationName] = newNotificationName
+		obc, err = k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Update(ctx, obc, metav1.UpdateOptions{})
+		// check whether bucket notification added
+		notificationPresent := utils.Retry(12, 2*time.Second, "notification is created for bucket", func() bool {
+			return helper.BucketClient.CheckBucketNotifications(namespace, storeName, obcName, bucketname, newNotificationName, s3endpoint)
+		})
+		assert.True(t, notificationPresent)
+		err = helper.NotificationClient.DeleteNotification(newNotificationName, topicName)
+		assert.Nil(t, err)
+		err = helper.TopicClient.DeleteTopic(newTopicName, storeName, httpEndpointService)
+		assert.Nil(t, err)
+	})
+
+	t.Run("create notifcation before topic, then add to it OBC", func(t *testing.T) {
+		reverseNotificationName := "reverse-notification"
+		reverseTopicName := "reverse-topic"
+		err := helper.NotificationClient.CreateNotification(reverseNotificationName, reverseTopicName)
+		assert.Nil(t, err)
+		created := utils.Retry(12, 2*time.Second, "notification is created", func() bool {
+			return helper.NotificationClient.CheckNotification(notificationName)
+		})
+		assert.True(t, created)
+
+		err = helper.TopicClient.CreateTopic(reverseTopicName, storeName, httpEndpointService)
```

Comment:
another interesting case would be:
- create notification
- create obc with notification label
- create topic

Reply:
Yeah I change last test case accordingly